### PR TITLE
[aggregator] Added e2e aggregator tests

### PR DESCRIPTION
**Pull request by georgemitenkov**
### Description

e2e tests for `Aggregator`.

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3301)

### CODE REVIEW

**Review by zekun000 — APPROVED** [review]
we should also add some new/delete aggregators and access random aggregators

**Review by gelash — APPROVED** [review]
please add tests that include some materialized ops, i.e. txns that reads & add / read & subtract.

File: aptos-move/e2e-move-tests/tests/aggregator.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                add(&mut h, &acc, i, i as u128)
+            }
+        })
+        .collect();
+    h.run_block(txns);
+
+    // Finally, check values.
+    let txns: Vec<SignedTransaction> = (0..block_size)
+        .map(|i| {
+            let case = i % 4;
```
Comment (sasha8) [inline aptos-move/e2e-move-tests/tests/aggregator.rs:69]: match?

File: aptos-move/e2e-move-tests/tests/aggregator.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    // Now test all operations. To do that, make sure aggregator have values large enough.
+    let txns: Vec<SignedTransaction> = (0..block_size)
+        .map(|i| add(&mut h, &acc, i, (i as u128) * 1000))
+        .collect();
+    h.run_block(txns);
+
+    // TODO: proptests with random transaction generator might be useful here.
+    let txns: Vec<SignedTransaction> = (0..block_size)
+        .map(|i| {
+            let case = i % 4;
```
Comment (sasha8) [inline aptos-move/e2e-move-tests/tests/aggregator.rs:52]: match?

File: aptos-move/e2e-move-tests/tests/aggregator.rs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+use language_e2e_tests::account::Account;
+
+mod common;
+
+fn setup() -> (MoveHarness, Account) {
+    initialize(common::test_dir_path("aggregator.data/pack"))
+}
+
+#[test]
+fn test_aggregators_e2e() {
```
Comment (sasha8) [inline aptos-move/e2e-move-tests/tests/aggregator.rs:21]: Should we add a test to check the delete logic? Say I try to add to a destroyed aggregator?

**Review by sasha8 — APPROVED** [review]
LGTM

File: aptos-move/e2e-move-tests/tests/aggregator.rs (hunk 3 above)
Comment (georgemitenkov) [reply aptos-move/e2e-move-tests/tests/aggregator.rs:21]: You cannot add to destroyed aggregator though, because the function consumes the value?

Comment (georgemitenkov) [reply aptos-move/e2e-move-tests/tests/aggregator.rs:21]: Actually I can still check if it is in the table or not

**Pull request merged**